Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs

```diff
@@ -44,356 +44,500 @@ public class ProjectController : Controller
-     * @api {get} api/Project/projects 
-     * @apiVersion 1.0.0
-     * @apiName Project
-     * @apiGroup Project
-     * @apiSuccessExample {json} Success-Response:
-     * HTTP/1.1 200 OK 
-     * {
-     *     "description":"Get List of Projects"
-     * }
-     */
+         * @api {get} api/project 
+         * @apiVersion 1.0.0
+         * @apiName GetProjectsAsync
+         * @apiGroup Project
+         * @apiParam {null} no parameter
+         * @apiSuccessExample {json} Success-Response:
+         * HTTP/1.1 200 OK 
+         * [
+          * {
+          *   "Name":"Slack",
+          *   "SlackChannelName":"SlackChannelName",
+          *   "IsActive":"True",
+          *   "TeamLeaderId":"34d1af3d-062f-4bcd-b6f9-b8fd5165e367",
+          *   "CreatedBy" : "Smith",
+          *   "CreatedDate" : "10/02/2016",
+          *   "UpdatedBy": "Smith",
+          *   "UpdatedDate" : "10/02/2016"
+          *   "TeamLeader": 
+          *   {
+          *         "Id":"34d1af3d-062f-4bcd-b6f9-b8fd5165e367",
+          *         "FirstName" : "John",
+          *         "Email" : "jone@promactinfo.com",
+          *         "LastName" : "Doe",
+          *         "SlackUserName" :"John",
+          *         "IsActive" : "True",
+          *         "JoiningDate" :"10-02-2016",
+          *         "NumberOfCasualLeave":0,
+          *         "NumberOfSickLeave":0,
+          *         "UniqueName":null,
+          *         "Role":null,
+          *         "Password":null,
+          *         "UserName": null
+          *         
+          *     } 
+          * }  
+          *]
+          *    
+         */
         [Authorize]
         [HttpGet]
-        [Route("getAllProjects")]
-        public async Task<IEnumerable<ProjectAc>> getAllProjects()
+        [Route("")]
+        public async Task<IActionResult> GetProjectsAsync()
         {
             try
             {
                 var user = await _userManager.FindByNameAsync(User.Identity.Name);
                 var userRole = await _userManager.IsInRoleAsync(user, "Employee");
-                _logger.LogInformation("UserRole Employee  "+userRole);
-                if (userRole == true)
+                _logger.LogInformation("UserRole Employee  " + userRole);
+                if (userRole)
                 {
                     _logger.LogInformation("call project repository for User");
-                    return await _projectRepository.GetAllProjectForUser(user.Id);
+                    return Ok(await _projectRepository.GetAllProjectForUserAsync(user.Id));
                 }
                 else
                 {
                     _logger.LogInformation("call project repository for projects");
-                    return await _projectRepository.GetAllProjects();
+                    return Ok(await _projectRepository.GetAllProjectsAsync());
                 }
             }
-            catch (Exception ex)
+            catch (FailedToFetchDataException)
```

Comment:
If some how we are not fetching any data form the database and database has no record yet then list are empty  that can be catch FailedToFetchDataException using of this we can show the end user list are empty. if my logic are wrong the i will remove it.